packages/engine/paima-funnel/src/reading.ts

```diff
@@ -40,12 +42,21 @@ export async function getBaseChainDataMulti(
   const resultList: ChainData[] = [];
   for (const block of blockData) {
     const blockEvents = events.filter(e => e.blockNumber === block.blockNumber);
+    console.log('block 1', block);
+
     const submittedData = await extractSubmittedData(
       blockEvents,
       block.timestamp,
       DBConn,
       caip2Prefix
     );
+
+    const batcherSubmittedData = submittedData.filter(s => s.fromBatcher);
+    if (batcherSubmittedData.length) {
+      const totalGasUsed = await getTotalGas(web3, blockEvents);
+      await splitGasCost(totalGasUsed, batcherSubmittedData, DBConn);
+    }
```

Comment:
I'm not sure if this is the correct place for this, I didn't want to bubble up more data for now (gas used)

Or should we move this closer to the STF loop, and write in the main loop as well?